endpoints/openrtb2/amp_auction_test.go

```diff
@@ -1929,3 +1933,68 @@ func TestSetTargeting(t *testing.T) {
 
 	}
 }
+
+func TestValidAmpResponseWhenRequestRejected(t *testing.T) {
+	reject := hookexecution.RejectError{
+		123,
+		hookexecution.HookID{ModuleCode: "foobar", HookCode: "foo"},
+		hooks.StageEntrypoint.String(),
+	}
+
+	testCases := []struct {
+		description         string
+		expectedAmpResponse AmpResponse
+		hookExecutor        hookexecution.HookStageExecutor
+	}{
+		{
+			"Assert correct AmpResponse when request rejected at entrypoint stage",
+			AmpResponse{Targeting: map[string]string{}},
+			rejectableHookExecutor{entrypointReject: &reject},
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.description, func(t *testing.T) {
+			actualAmpObject := analytics.AmpObject{}
+			logger := newMockLogger(&actualAmpObject, nil)
+			stored := map[string]json.RawMessage{
+				"1": json.RawMessage(validRequest(t, "site.json")),
+			}
+
+			deps := &endpointDeps{
+				fakeUUIDGenerator{},
+				&nobidExchange{},
+				mockBidderParamValidator{},
+				&mockAmpStoredReqFetcher{data: stored},
+				empty_fetcher.EmptyFetcher{},
+				empty_fetcher.EmptyFetcher{},
+				&config.Configuration{MaxRequestSize: maxSize},
+				&metricsConfig.NilMetricsEngine{},
+				logger,
+				map[string]string{},
+				false,
+				[]byte{},
+				openrtb_ext.BuildBidderMap(),
+				nil,
+				nil,
+				hardcodedResponseIPValidator{response: true},
+				empty_fetcher.EmptyFetcher{},
+				test.hookExecutor,
+			}
+
+			req := httptest.NewRequest("POST", "/openrtb2/amp?tag_id=1", nil)
+			recorder := httptest.NewRecorder()
+
+			deps.AmpAuction(recorder, req, nil)
+			assert.Equal(t, recorder.Code, http.StatusOK, "Endpoint should return 200 OK.")
```

Comment:
According to the spec, a rejection at the entrypoint stage should result in a 204:

> For /openrtb2/amp and /setuid, modules must be able to modify the GET query string or reject the request entirely. Rejection results in HTTP 204 and a metric associating the rejection with the module.

In that case, the body should be empty if using 204.

Reply:
Same here